File: modules/remotebackend/regression-tests/http-backend.rb
#!/usr/bin/env ruby
require "rubygems"
require 'bundler/setup'
require "webrick"
$:.unshift File.dirname(__FILE__)
require "dnsbackend"
require "backend"
require "pathname"

server = WEBrick::HTTPServer.new(
	:Port=>62434,
	:BindAddress=>"localhost",
#	Logger: WEBrick::Log.new("remotebackend-server.log"),
	:AccessLog=>[ [ File.open("remotebackend-access.log", "w"), WEBrick::AccessLog::COMBINED_LOG_FORMAT ] ] 
)

be = Handler.new(Pathname.new(File.join(File.dirname(__FILE__),"remote.sqlite3")).realpath.to_s)
server.mount "/dns", DNSBackendHandler, be
server.mount_proc("/ping"){ |req,resp| resp.body = "pong" }

trap('INT') { server.stop }
trap('TERM') { server.stop }

server.start

File: modules/remotebackend/regression-tests/http-backend.py
#!/usr/bin/env python

import http.server
from backend import BackendHandler
from dnsbackend import DNSBackendHandler
import os

class DNSBackendServer(http.server.HTTPServer):
    def __init__(self, *args, **kwargs):
        path = os.path.dirname(os.path.realpath(__file__))
        self.handler = BackendHandler(options={'dbpath': os.path.join(path, 'remote.sqlite3')})
        super().__init__(*args, **kwargs)

    def finish_request(self, request, client_address):
        """Finish one request by instantiating RequestHandlerClass."""
        h = self.RequestHandlerClass(request, client_address, self, handler=self.handler)

def main():
    server = DNSBackendServer(('', 62434), DNSBackendHandler)
    try:
        server.serve_forever()
    except KeyboardInterrupt:
        pass

main()
